common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesAuthorityValidationMetadataCache.java

```diff
@@ -58,20 +57,31 @@ public SharedPreferencesAuthorityValidationMetadataCache(final Context context)
                 );
     }
 
-    @Override
-    public String generateAuthorityValidationMetadataKey(final String environment) {
+    /**
+     * Generate cache Key for authority validation metadata.
+     *
+     * @param environment Environment
+     * @return String
+     */
+    private String generateAuthorityValidationMetadataKey(@NonNull final String environment) throws ClientException {
+        if (null == environment) {
+            throw new ClientException(ENVIRONMENT_CANNOT_BE_NULL_AS_A_AUTHORITY_VALIDATION_METADATA_KEY);

```

Comment:
Given that you have defined `@NonNull` as a requirement of this API, I would suggest sparing yourself the extra work of declaring the checked exception and instead throw an `IllegalArgumentException` which, because it is a subclass of `RuntimeException` will allow you skip declaring it on the method signature